src/sp_wrapper.c

```diff
@@ -0,0 +1,31 @@
+#include "php_snuffleupagus.h"
+#include "sp_config.h"
+
+ZEND_DECLARE_MODULE_GLOBALS(snuffleupagus)
+
+void sp_disable_wrapper() {
+  HashTable *orig = php_stream_get_url_stream_wrappers_hash();
+  HashTable *full = pemalloc(sizeof(*full), 1);
+  zval *zv;
+  zend_string *zs;
+
+  zend_hash_init(full, zend_hash_num_elements(orig), NULL, NULL, 1);
+  zend_hash_copy(full, orig, NULL);
```

Comment:
What about using `zend_hash_del` to remove every entry that isn't in the whitelist instead?

Reply:
I'd have to iterate through every entry as well, and i'm not _sure_ about the comportment of `ZEND_HASH_FOREACH` when element are removed from the current hashtable in the loop.

Reply:
Fine by be, please add a comment in the code to explain this decision :)